### PR TITLE
Use JSON schema for settings.

### PR DESCRIPTION
**Pull request by afshin**

### CODE REVIEW

File: packages/settingeditor-extension/src/settingeditor.ts [hunk 1]
```diff
@@ -604,64 +600,92 @@ namespace Private {
...
+    // Finally, use the defaults from the registry schema.
+    if (!hint) {
+      hint = registry.schema.properties[key].default;
+    }
+
+    return typeof hint === 'string' ? hint : '';
+  }
+
+  /**
+   * Populate the fieldset with a specific plugin's metaata.
```
Comment (blink1073) [inline packages/settingeditor-extension/src/settingeditor.ts]: metadata*

File: packages/settingeditor-extension/src/settingeditor.ts [hunk 2]
```diff
@@ -604,64 +600,92 @@ namespace Private {
...
+   * #### Notes
+   * The order of priority for overridden hints is as follows, from most
+   * important to least:
+   * 1. Data set by the end user in a settings file.
+   * 2. Data set by the plugin author as a schema default.
+   * 3. Data set by the plugin author as a top-level key of the schema.
+   */
+  function getHint(key: string, registry: ISettingRegistry, plugin: ISettingRegistry.IPlugin): string {
+    // First, give priorty to checking if the hint exists in the user data.
+    let hint = plugin.data.user[key];
```
Comment (blink1073) [inline packages/settingeditor-extension/src/settingeditor.ts]: Should be `plugin.data.user && plugin.data.user[key];`, and `plugin.data.composite && plugin.data.composite[key];` below.

Comment (afshin) [inline packages/settingeditor-extension/src/settingeditor.ts]: I addressed this upstream.

**Comment by blink1073** [comment]
Looks great, thanks!

**Pull request merged**